build/Publish.targets

```diff
@@ -112,16 +112,23 @@
         <Overwrite>true</Overwrite>
       </FilesToPublish>
 
-      <PackagesToPublish Include="$(DependencyPackagesDir)*.nupkg" Exclude="$(DependencyPackagesDir)*.symbols.nupkg" />
-      <PackagesToPublish Include="$(DependencyPackagesDir)*.symbols.nupkg" Exclude="@(PackagesToPublish)" IsSymbolsPackage="true" />
+      <_SymbolArtifactPath Include=">$(DependencyPackagesDir)%(PackageArtifact.Identity).*$(BuildNumber).nupkg" Artifact="%(PackageArtifact.Identity)" />
+      <_PackageArtifactPath Include=">$(DependencyPackagesDir)%(PackageArtifact.Identity).*$(BuildNumber).nupkg" Exclude="@(_SymbolArtifactPath)" Artifact="%(PackageArtifact.Identity)" />
+
+      <PackagesToPublish Include="@(PackageArtifact)">
+        <Category>%(PackageArtifact.Category)</Category>
+        <PackagePath>@(_PackageArtifactPath->WithMetadataValue('Artifact', %(Identity)))</PackagePath>
+        <SymbolPackagePath>@(_SymbolArtifactPath->WithMetadataValue('Artifact', %(Identity)))</SymbolPackagePath>
+      </PackagesToPublish>
     </ItemGroup>
 
     <Message Text="@(FilesToPublish -> '%(FullPath) -> %(RelativeBlobPath)','%0A')" />
   </Target>
 
   <Target Name="_CheckFilesToPublish">
     <ItemGroup>
-      <_MissingFiles Include="%(FilesToPublish.Identity)" Condition=" ! Exists(%(FilesToPublish.Identity))" />
+      <_MissingFiles Include="%(FilesToPublish.Identity)" Condition="!Exists(%(FilesToPublish.Identity))" />
+      <_MissingFiles Include="%(PackagesToPublish.PackagePath)" Condition="!Exists(%(PackagesToPublish.PackagePath)) AND '%(PackagesToPublish.Category)' == 'ship'" />
```

Comment:
My guess is I shouldn't check the category here. The packages dir should have all the packages, no?

Reply:
Yes, it should. We still want to publish shipoob and noship packages to the transport feed - they might be used by downstream partners, even if they aren't intended for nuget.org